tensorflow/contrib/tensorrt/convert/convert_nodes.cc

```diff
@@ -75,13 +102,108 @@ inline tensorflow::Status ConvertDType(tensorflow::DataType tf_dtype,
     case tensorflow::DataType::DT_HALF:
       *trt_dtype = nvinfer1::DataType::kHALF;
       break;
+#if NV_TENSORRT_MAJOR > 3
+    case tensorflow::DataType::DT_INT32:
+      *trt_dtype = nvinfer1::DataType::kINT32;
+      break;
+#endif
     default:
       return tensorflow::errors::InvalidArgument(
           "Unsupported data type " + tensorflow::DataTypeString(tf_dtype));
   }
   return tensorflow::Status::OK();
 }
 
+// return whether or not the broadcast is feasible;
+bool TensorRTGetBroadcastShape(const nvinfer1::Dims& operand_l,
+                               const bool operand_l_is_tensor,
+                               const nvinfer1::Dims& operand_r,
+                               const bool operand_r_is_tensor,
+                               nvinfer1::Dims* operand_l_new_shape,
+                               nvinfer1::Dims* operand_r_new_shape) {
+  /*******************************************************************************
+    TensorRT Elementwise op supports broadcast but requires both tensor to be of
+    Identical rank
+
+    We consider case of: i. Tensor op Const; ii. Tensor op Tensor
+    note: const op const (constant folding) should fallback to TensorFlow
+
+    broadcast scheme:
+    T: 1 3 5          (tensor would not have batch dimension)
+    W: 1 1 3 1        (weight would have all explicit dimensions)
+    i. fill in explicit dimensions
+    -> T: -1 1 3 5  (we put a -1 for batch dimension)
+    -> W:  1 1 3 1
+    ii. compare broadcast feasibility
+
+    we cannot support these since TensorRT does not allow manipulation on batch
+  dimension, we cannot generate output with proper shape
+    T: 3 5 1
+    W: 1 1 1 1 3 5 1
+    -> T: 1 1 1 -1 3 5 1
+    -> W: 1 1 1  1 3 5 1
+  *******************************************************************************/
+  static const int max_nb_dims = nvinfer1::Dims::MAX_DIMS + 1;
+  const size_t element_size = sizeof(operand_l.d[0]);
+
+  // fill in dimensions
+  int l_s[max_nb_dims];
+  std::fill(l_s, l_s + max_nb_dims, 1);
+  int l_d = operand_l_is_tensor ? operand_l.nbDims + 1 : operand_l.nbDims;
+  int r_s[max_nb_dims];
+  std::fill(r_s, r_s + max_nb_dims, 1);
+  int r_d = operand_r_is_tensor ? operand_r.nbDims + 1 : operand_r.nbDims;
+
+  int max_d = std::max(l_d, r_d);
+  std::memcpy(l_s + max_d - operand_l.nbDims, operand_l.d,
+              operand_l.nbDims * element_size);
+  std::memcpy(r_s + max_d - operand_r.nbDims, operand_r.d,
+              operand_r.nbDims * element_size);
+
+  // set -1 for batch dimension, since batch size is not supposed to be
+  // broadcasted
+  if (operand_l_is_tensor) {
+    if (max_d != l_d) {  // if broadcast beyond batch dimension, fail
+      return false;
+    }
+    l_s[0] = -1;
```

Comment:
It seems this does not align with what the comment says:
```
+    T: 3 5 1
+    -> T: 1 1 1 -1 3 5 1
```
but here it assigns -1 to first element.

Reply:
The comment is trying to illustrate why we cannot support this case.
In the code, if we need to promote the rank of the tensor operand, the state if (max_d != l_d) returns true and the function returns false.